teaser/examples/e11_export_besmod_models.py

```diff
@@ -0,0 +1,195 @@
+        2024: 35 + 273.15
+    }
+
+    # In the examples, the parameters for BESMod.Systems.UserProfiles.TEASERProfiles are configured,
+    # including internal gains and heating profiles for each zone.
+    # BESMod requires 24-hour heating profiles, which are used
+    # to define the parameters of the setBakTSetZone Pulse block.
+    # By default, the TEASER profiles are applied, but these can be customized if needed.
+
+    # Additionally, location-specific parameters must be set, which can be achieved using the following function.
+    # The default values provided here correspond to Mannheim.
+    weather_file_path = utilities.get_full_path(
+        os.path.join(
+            "data",
+            "input",
+            "inputdata",
+            "weatherdata",
+            "DEU_BW_Mannheim_107290_TRY2010_12_Jahr_BBSR.mos"))
+
+    prj.set_location_parameters(t_outside=262.65,
+                                t_ground=286.15,
+                                weather_file_path=weather_file_path,
+                                calc_all_buildings=True)
+
+    # To make sure the parameters are calculated correctly we recommend to
+    # run prj.calc_all_buildings() function which is here already done in the set_location_parameters function.
+
+    # Export all buildings to BESMod and include them in predefined example systems.
+    path = prj.export_besmod(
+        THydSup_nominal=THydSup_nominal,
+        path=save_dir_path,
+        examples=examples
+    )
+
+    # The partial retrofit option of the energy system in BESMod can also be utilized.
+    # To enable this, the nominal heat flow of each zone in the building must be extracted prior to the retrofit.
+    QBuiOld_flow_design = {
+        bldg.name: {
+            tz.name: tz.model_attr.heat_load for tz in bldg.thermal_zones
+        }
+        for bldg in prj.buildings
+    }
+
+    # Retrofit project buildings and recalculate parameters.
+    prj.name = "ArchetypeExample_partial_retrofit"
+    prj.retrofit_all_buildings(
+        year_of_retrofit=2015,
+        type_of_retrofit="adv_retrofit",
+        window_type='Alu- oder Stahlfenster, Isolierverglasung',
+        material='EPS_perimeter_insulation_top_layer'
+    )
+    prj.calc_all_buildings()
+
+    # By default, radiator transfer systems are not retrofitted when the
+    # QBuiOld_flow_design parameter is provided and differs from the new nominal heat flow.
+    # Additionally, new THydSup_nominal temperatures can be specified alongside
+    # THydSupOld_design values, which are used for radiator sizing but not for control settings.
+
+    path = prj.export_besmod(
+        THydSup_nominal=THydSup_nominal,
+        QBuiOld_flow_design=QBuiOld_flow_design,
+        path=save_dir_path,
+        examples=examples
+    )
+
+    # Additionally, we have the flexibility to define custom templates for including buildings in specific setups.
+    # For instance, a custom template is defined here to include the building in the
```

Comment:
I was thinking if having three examples: export, retrofit, templates for BESMod would be "better" in the sense that it does not overload users. Then again, the examples folder would get really big. What do you think?

Reply:
I would keep it in one example because they build upon each other. But maybe I could make it more clear that they could be done separately or insert if statements for just running basic export, partial retrofit export and custom export.